### PR TITLE
exa: Fix build for arm64, add doc for macOS 10.12

### PR DESCRIPTION
**Pull request by judaew**
Changes:
* Replace `pandoc` by `go-md2man`. `pandoc` cannot be build because it dependency stack only support the arch x86_64. It's maybe fix build for arm64.
* Man-pages now be available on macOS 10.12 and below.

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 11.5.2 20G95 x86_64
Xcode 12.5.1 12E507

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Review by harens — COMMENTED** [review]
Thanks @judaew for not only adding the amazing  go-md2man port, but for implementing this bugfix! 🎉 My main comment is the following warning during the build:

```
WARNING: go-md2man does not handle node type HTMLBlock
```

This results in things like the following:

image

...which isn't ideal, since this isn't an issue with pandoc. At the same time, it's not a dealbreaker either compared with not providing the man pages in the first place for certain systems.

I think the best solution is that go-md2man should be used where pandoc can't, and we accept that there are some minor formatting errors on those systems. Alternatively, we could look for a different program that does support code blocks. Thoughts?

Either way, thanks @judaew for sending this PR. Greatly appreciated :) 👍

File: sysutils/exa/Portfile [hunk 1]
```diff
@@ -40,20 +40,15 @@ build.pre_args-replace   --frozen --offline
 variant git description {Build with git functionality} {}
 
 variant doc description {Build man pages} {
-    depends_build-append port:pandoc
+    depends_build-append port:go-md2man
     post-build {
-        system -W ${worksrcpath} "${prefix}/bin/pandoc --standalone -f markdown -t man man/exa.1.md > man/exa.1"
-        system -W ${worksrcpath} "${prefix}/bin/pandoc --standalone -f markdown -t man man/exa_colors.5.md > man/exa_colors.5"
+        system -W ${worksrcpath}/man "go-md2man -in=${name}.1.md -out=${name}.1"
+        system -W ${worksrcpath}/man "go-md2man -in=${name}_colors.5.md \
```
[the remark this answers is not in the document]
Comment (judaew) [reply sysutils/exa/Portfile]: Thanks so much, I remember it!

**Comment by judaew** [comment]
Usually, man pages written in markdown format don't use HTML tags. So go-md2man, ronn, or ronn-ng will do fine, but not in this case.

@harens I agree with you and also think the best solution is to use go-md2man for macOS 10.12 and below or Apple Silicon. I made the change as you requested. Can you check it?

File: sysutils/exa/Portfile [hunk 2]
```diff
@@ -40,20 +40,32 @@ build.pre_args-replace   --frozen --offline
...
+        post-build {
+            system -W ${worksrcpath}/man "${prefix}/bin/go-md2man \
+                -in=${name}.1.md -out=${name}.1"
+            system -W ${worksrcpath}/man "${prefix}/bin/go-md2man \
+                -in=${name}_colors.5.md -out=${name}_colors.5"
+        }
+        notes "
+            The man pages for macOS 10.12 and below or Apple Silicon on may \
+            have minor formatting errors as they were built with go-md2man \
+            instead of pandoc.
```
Comment (harens) [inline sysutils/exa/Portfile:55]: This is good, I like these notes. Just a minor nitpicky typo with _"Apple Silicon on may"_ and then it LGTM.

Comment (judaew) [reply sysutils/exa/Portfile]: Thanks, that's better? I am not a native English speaker and therefore have some difficulties.

**Review by harens — APPROVED** [review]
Thanks @judaew for this PR!

I hope you don't mind, but I just realised that the notes will have double spacing (e.g. https://github.com/macports/macports-ports/pull/11604#discussion_r671569886), so to speed things up I thought I'd correct it quickly.

Thank you so much for your work on this. 👍

File: sysutils/exa/Portfile (hunk 2 above)
Comment (harens) [reply sysutils/exa/Portfile]: That's perfect. Thanks.

**Comment by judaew** [comment]
Thanks @harens 🌟

**Pull request merged**